.github/workflows/ci.yml

```diff
@@ -90,37 +75,53 @@ jobs:
       env:
         COLOR: 'yes'
       run: |
-        make test-only
+        make test
         python -m coverage xml
     - name: Upload coverage
       uses: codecov/codecov-action@v3
       with:
         file: ./coverage.xml
         flags: unit
-        fail_ci_if_error: false
+
+  check:  # This job does nothing and is only used for the branch protection
+    if: always()
+
+    needs: [lint, test]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@release/v1
+      with:
+        jobs: ${{ toJSON(needs) }}
 
   deploy:
     name: Deploy
+    environment: release
     runs-on: ubuntu-latest
-    needs: test
-    # Run only on pushing a tag
+    needs: [check]
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
     steps:
     - name: Checkout
       uses: actions/checkout@v3
-    - name: Setup Python 3.8
+    - name: Setup Python
       uses: actions/setup-python@v4
       with:
         python-version: 3.8
     - name: Install dependencies
       run:
-        python -m pip install -U pip wheel twine build
-    - name: Make dists
-      run:
-        python -m build
-    - name: PyPI upload
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
```

Comment:
@Dreamsorcerer there's a repo-global secret with this name in the repo, but there's none in environments. I suppose, nobody migrated it. I think I could generate a new token in the bot account. Currently, I can't set up Trusted Publishing for this repo since I don't have Owner access in the PyPI project. I wonder if @hellysmile has said access...